casanovo/utils.py

```diff
@@ -251,3 +254,34 @@ def log_sequencing_report(
         logger.info(
             "Median Peptide Length: %d", run_report["median_sequence_length"]
         )
+
+
+def check_dir(
+    dir: pathlib.Path, file_patterns: Iterable[re.Pattern[str]]
+) -> None:
+    """
+    Check that no file names in dir match any of file_patterns
+
+    Parameters
+    ----------
+    dir : pathlib.Path
+        The directory to check for matching file names
+    file_patterns : Iterable[re.Pattern[str]]
+        File name re patterns to test file names against
+
+    Raises
+    ------
+    FileExistsError
+        If matching file name is found in dir
+    """
+    for pattern in file_patterns:
+        comp_pattern = re.compile(pattern)
+        for file in dir.iterdir():
+            if not file.is_file():
+                continue
+
+            if comp_pattern.fullmatch(file.name) is not None:
```

Comment:
Ok, now I see that this is not necessary for the log and mztab check, but it is for the checkpoints check. However, maybe [glob](https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob) is a better fit.

Reply:
I took a look at glob and it looks like the regex that glob uses is limited compared to the Python re module. For example, I don't think it would be possible to create regex that matches the checkpoint file names exactly with glob. It would be possible to create a pattern (`<output_root>.epoch=*step=*.ckpt` or similar) that matches a superset of checkpoint names, for example it would match `<output_root>.epoch=foo-step=bar.ckpt`. Maybe the decrease in code complexity would be worth the tradeoff in this case though.

Reply:
Yes, it's indeed a bit more limited functionality, but sufficient for what we want to do imo.